src/main/java/com/marginallyclever/convenience/ToggleButtonIcon.java

```diff
@@ -0,0 +1,55 @@
+package com.marginallyclever.convenience;
+
+import com.marginallyclever.makelangelo.Translator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A button with multiple sets of icons, texts and optionally foreground colors
+ */
+public class ToggleButtonIcon extends JButton {
+    private static final Logger logger = LoggerFactory.getLogger(ToggleButtonIcon.class);
+
+    private final List<String> translations;
+    private final List<ImageIcon> icons;
+    private final List<Color> foregroundColors;
```

Comment:
are these three lists always the same length?

Reply:
yes. `foregroundColors` is optionnal

Reply:
if they were all the same length I would say make a single list of a new subclass.  if foreground color is optional then don't bother.